npm/networkPolicyController.go

```diff
@@ -351,10 +351,24 @@ func (c *networkPolicyController) syncAddAndUpdateNetPol(netPolObj *networkingv1
 			return fmt.Errorf("[syncAddAndUpdateNetPol] Error: creating ipset named port %s with err: %v", set, err)
 		}
 	}
-	for _, list := range lists {
-		if err = ipsMgr.CreateList(list); err != nil {
-			return fmt.Errorf("[syncAddAndUpdateNetPol] Error: creating ipset list %s with err: %v", list, err)
+
+	// lists is a map with list name and members as value
+	// NPM will create the list first and increments the refer count
+	for listKey, _ := range lists {
+		if err = ipsMgr.CreateList(listKey); err != nil {
+			return fmt.Errorf("[syncAddAndUpdateNetPol] Error: creating ipset list %s with err: %v", listKey, err)
+		}
+		ipsMgr.IpSetReferIncOrDec(listKey, util.IpsetSetListFlag, ipsm.IncrementOp)
```

Comment:
better to include this call in CreateList(). So, we do not need to expose this function to other packages and encapsulate information in ipsMgr.

Reply:
As i explained above, this should not be used in createList, but only when netowrkPolicy or a 2nd level List uses an IPset

Reply:
In this case, I prefer to having another function which functions "CreateList and IPSetReferIncOrDec". It will be easy to remember and help to encapsulate details. 
I assume that your next PR is about those reference counts, we can revisit this in the PR.

Reply:
I agree, in refer count work item, we can look at different scenarios and better organize these functions.